### DISCUSSION TITLE
Grid Monthly Energy Funny Values

### DISCUSSION THREAD

**Discussion by thinuspollard** [kellerza/sunsynk#357] [General]
Hi All

Sunsynk 8kW inverter.

According to the docs and code modbus register 67 contains the Monthly Grid Energy value. Reading it via Home Assistant (via this add-on) provides a value that is not in line with reality.

The dongle (via website) reports around 15kWh imported for November, but the value from the add-on is around 58kWh. So no relation at all.

I suspect that the register provides an incorrect value, and the dongle+website adjusts on other values. I'm assuming the add-on just passes on the value it reads (adjusting the scale as documented). I have not exhaustivly checked the other values for sanity, but this stood out to me.

Question, has anyone else observed this? Firmware issue?

Edit: The datetime on the inverter is correct for my timezone

Thanks
Thinus

**Comment by kellerza** [comment]
Hi Thinus, personally I don't use these, so it might not be correct. 

Maybe the app reports the sum of some registers? The translated version of the modbus register document is not always accurate.

They are defined here for single phase
https://github.com/kellerza/sunsynk/blob/main/src/sunsynk/definitions/single_phase.py#L124

**Reply by thinuspollard** [reply]
HI @kellerza 
Thanks for all your effort, love the add-on

As mentioned, the code matches the docs that I have been able to find, values read from the register seems wrong. The other monthly figures look sane when compared with the SunSynk app.

I think I will log a support issue with SunSynk and see if they have comments.
